server/py/framework/db/sqldb/db.py

```diff
@@ -1267,6 +1262,12 @@ def validate_artifact_removal_preconditions(
                 f"{db_artifact.iteration}, producer_id {db_artifact.producer_id} and {db_artifact.uid} uid. "
                 f"The artifact is used by {dependent_endpoints_count} endpoints"
             )
+        if db_artifact.child_artifacts:
+            raise mlrun.errors.MLRunConflictError(
+                f"Failed deleting artifact {db_artifact.key} in project {db_artifact.project}, iteration "
+                f"{db_artifact.iteration}, producer_id {db_artifact.producer_id} and {db_artifact.uid} uid. "
+                f"The artifact has {len(db_artifact.child_artifacts)} child artifacts, please delete them first"
```

Comment:
We do not use please in error messages.
```suggestion
                f"The artifact has {len(db_artifact.child_artifacts)} child artifacts. Delete them before proceeding."
```